component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -380,7 +382,8 @@ private void processSMSOTPMandatoryCase(AuthenticationContext context, HttpServl
         //the authentication flow happens with sms otp authentication.
         String tenantDomain = context.getTenantDomain();
         String errorPage = getErrorPage(context);
-        if (context.isRetrying() && !Boolean.parseBoolean(request.getParameter(SMSOTPConstants.RESEND))) {
+        if (context.isRetrying() && !Boolean.parseBoolean(request.getParameter(SMSOTPConstants.RESEND))
+                && !isMobileNumberUpdateFailed(context)) {
```

Comment:
shall we add a debug log here explaining the scenario along with necessary context information e.g. username, tenant domain

Reply:
Added debug log